hexrd/ui/image_load_manager.py

```diff
@@ -336,7 +336,13 @@ def add_omega_metadata(self, ims_dict):
             ims_dict[key].metadata['omega'] = omw.omegas
 
     def get_range(self, ims):
-        return range(len(ims))
+        start = 0
+        if len(ims) != self.data.get('total_frames', [len(ims)])[0]:
+            # In the case of hdf5 and npz files we need to handle empty
+            # frames via the ProcessedImageSeries 'frame_list' arg to slice
+            # the frame list
+            start = self.data.get('empty_frames', 0)
+        return range(start, len(ims))
```

Comment:
So I think we need to replace `len(ims)` with the `steps` values from the omega metadata table.  This will also alow use to trim frames off the end as well...  right?

Reply:
we'll come back to this if need be...

Reply:
@bnmajor -- what do you think?

Reply:
@joelvbernier range creates the list of frames to use - in this case that list of frames starts at the value of `empty-frames` and ends at the last frame of the imageseries for the `hdf5` and `npz` files. The only way to specify a different value for the max frames per file or max total frames is through the image stack dialog right now, and the only way to load `hdf5` and `npz` files is through the load panel. Since this check only handles the `hdf5` and `npz` files I think that setting the stop value as the last frame makes sense right now (since the user can no longer change that value in the table).

If we add the option to change the number of steps in the load panel or add support for loading `npz` or `hdf5` files through the image stack dialog then yes, this range will need to be calculated differently. Let me know if I'm misunderstanding the use case you're thinking of